cmd/gogio/windowsbuild.go

```diff
@@ -264,53 +240,158 @@ func (f *windowsManifest) encode(w io.Writer) error {
+		windowsInfoValueFixed{
+			Signature:      0xFEEF04BD,
+			StructVersion:  0x00010000,
+			FileVersion:    v.Version,
+			ProductVersion: v.Version,
+			FileFlagMask:   0x3F,
+			FileFlags:      0,
+			FileOS:         0x40004,
+			FileType:       0x1,
+			FileSubType:    0,
+		},
+		// https://docs.microsoft.com/pt-br/windows/win32/menurc/stringfileinfo
+		newValue(valueText, "StringFileInfo", []io.WriterTo{
+			// https://docs.microsoft.com/pt-br/windows/win32/menurc/stringtable
+			newValue(valueText, fmt.Sprintf("%04X%04X", v.Language, page), []io.WriterTo{
+				// https://docs.microsoft.com/pt-br/windows/win32/menurc/string-str
+				newValue(valueText, "ProductVersion", v.VersionHuman),
+				newValue(valueText, "FileVersion", v.VersionHuman),
+				newValue(valueText, "FileDescription", v.Name),
+				newValue(valueText, "ProductName", v.Name),
+				// TODO include more data: gogio must have some way to provide such information (like Company Name, Copyright...)
+			}),
+		}),
+		// https://docs.microsoft.com/pt-br/windows/win32/menurc/varfileinfo
+		newValue(valueBinary, "VarFileInfo", []io.WriterTo{
+			// https://docs.microsoft.com/pt-br/windows/win32/menurc/var-str
+			newValue(valueBinary, "Translation", uint32(page)<<16|uint32(v.Language)),
+		}),
+	})
+
+	// For some reason the ValueLength of the VS_VERSIONINFO must be the byte-length of the `windowsInfoValueFixed`:
+	t.ValueLength = 52
+
+	var verrsrc bufferCoff
+	if _, err := t.WriteTo(&verrsrc); err != nil {
 		return err
 	}
 
-	return template.Execute(w, f)
+	b.Coff.AddResource(16, 1, &verrsrc)
+
+	return nil
+}
+
+type windowsInfoValueFixed struct {
+	Signature      uint32
+	StructVersion  uint32
+	FileVersion    [2]uint32
+	ProductVersion [2]uint32
+	FileFlagMask   uint32
+	FileFlags      uint32
+	FileOS         uint32
+	FileType       uint32
+	FileSubType    uint32
+	FileDate       [2]uint32
+}
+
+func (v windowsInfoValueFixed) WriteTo(w io.Writer) (_ int64, err error) {
+	return 0, binary.Write(w, binary.LittleEndian, v)
+}
+
+type windowsInfoValue struct {
+	Length      uint16
+	ValueLength uint16
+	Type        uint16
+	Key         []byte
+	Value       []byte
+}
+
+func (v windowsInfoValue) WriteTo(w io.Writer) (_ int64, err error) {
+	// binary.Write doesn't support []byte inside struct.
+	if err = binary.Write(w, binary.LittleEndian, [3]uint16{v.Length, v.ValueLength, v.Type}); err != nil {
+		return 0, err
+	}
+	if _, err = w.Write(v.Key); err != nil {
+		return 0, err
+	}
+	if _, err = w.Write(v.Value); err != nil {
+		return 0, err
+	}
+	return 0, nil
+}
+
+const (
+	valueBinary uint16 = 0
+	valueText   uint16 = 1
+)
+
+func newValue(valueType uint16, key string, input interface{}) windowsInfoValue {
+	v := windowsInfoValue{
+		Type: valueType,
+		Key:  utf16Encode(key),
+	}
+
+	v.Length = 6 + uint16(len(v.Key))
+	v.Key = append(v.Key, make([]byte, v.Length%4)...) // Padding, it must be 32-bit.
```

Comment:
The pading calculation is only correct if v.Length%2 == 0. The correct calculation is (4 - length%4).